src/confcom/azext_confcom/security_policy.py

```diff
@@ -935,19 +935,38 @@ def load_policy_from_virtual_node_yaml_str(
             # mounts
             mounts = copy.deepcopy(config.DEFAULT_MOUNTS_VIRTUAL_NODE)
             volumes = case_insensitive_dict_get(spec, "volumes") or []
-            configmap_volume_names = [
-                case_insensitive_dict_get(volume, "name") if "configmap" in volume else None for volume in volumes
-            ]
+
+            # set of volume types that are read-only by default
+            read_only_types = {"configMap", "secret", "downwardAPI", "projected"}
+
             volume_mounts = case_insensitive_dict_get(container, "volumeMounts")
             if volume_mounts:
                 for mount in volume_mounts:
+                    mount_name = case_insensitive_dict_get(mount, "name")
+                    mount_path = case_insensitive_dict_get(mount, "mountPath")
+
+                    # find the corresponding volume
+                    volume = next((vol for vol in volumes if case_insensitive_dict_get(vol, "name") == mount_name), None)
+
+                    # determine if this volume is one of the read-only types
+                    read_only_volume_type = next(iter(read_only_types.intersection(volume.keys())), None)
+
+                    if read_only_volume_type:
+                        # log warning if readOnly is explicitly set to false for a read-only volume type
+                        if case_insensitive_dict_get(mount, "readOnly") is False:
+                            logger.warning(
+                                "%s volume '%s' in container '%s' is read-only by default, but readOnly: false was specified. Setting readOnly: true for policy generation.",
```

Comment:
I'd reword this to say the volume *can't* be read/write. Not just that it differs from the default

Reply:
gotcha, makes sense. done!